oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1398,6 +1399,24 @@ private void delayOnModifications(final long durationMs, final AtomicBoolean can
         }
     }
 
+    public UpdateOp collectGarbageOnDocument(DocumentNodeStore store, NodeDocument doc) {
```

Comment:
I'm working on this approach, will commit the changes soon.

Reply:
After some work in that area, I'm abandoning this approach. I initially thought it would be a good idea because that method is already quite long, so splitting it could improve its readability. However, the _real core_ part would be [just these lines](https://github.com/apache/jackrabbit-oak/blob/cf870b532b107b0c243e1d0180f1213d3f6e9197/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java#L752-L756), as the rest of the code is essentially to keep the progress of the iterators.

So I simply [added this part](https://github.com/apache/jackrabbit-oak/blob/cf870b532b107b0c243e1d0180f1213d3f6e9197/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java#L768-L772) to the facade method I already added.